lib/shelley/src/Cardano/Wallet/Shelley/MinimumUTxO.hs

```diff
@@ -66,11 +80,18 @@ computeMinimumCoinForUTxOShelleyBasedEra
     -> TokenMap
     -> Coin
 computeMinimumCoinForUTxOShelleyBasedEra
-    (MinimumUTxOForShelleyBasedEra era pp) addr tokenMap =
-        unsafeCoinFromCardanoApiCalculateMinimumUTxOResult $
-        Cardano.calculateMinimumUTxO era
-            (embedTokenMapWithinPaddedTxOut era addr tokenMap)
-            (Cardano.fromLedgerPParams era pp)
+    (MinimumUTxOForShelleyBasedEra era pp) addr tokenMap = case era of
+        -- Here we treat the Babbage era specially and use the ledger
+        -- to compute the minimum ada quantity, bypassing the Cardano
+        -- API. This appears to be significantly faster.
+        Cardano.ShelleyBasedEraBabbage ->
+            computeLedgerMinimumCoinForBabbage pp addr
+                (TokenBundle txOutMaxCoin tokenMap)
+        _ ->
+            unsafeCoinFromCardanoApiCalculateMinimumUTxOResult $
+            Cardano.calculateMinimumUTxO era
+                (embedTokenMapWithinPaddedTxOut era addr tokenMap)
```

Comment:
Don't think it matters and not sure if there is an obvious fix but FYI: perhaps slightly confusing that we're manually constructing `TokenBundle txOutMaxCoin` in one branch and calling `embedTokenMapWithinPaddedTxOut` in another branch.

Reply:
> Don't think it matters and not sure if there is an obvious fix but FYI: perhaps slightly confusing that we're manually constructing TokenBundle txOutMaxCoin in one branch and calling embedTokenMapWithinPaddedTxOut in another branch.

I agree it's not ideal!

I'd definitely like to tidy this up in a later PR, if possible.

Perhaps it won't be necessary to tidy though if we can work out why the slowdown happens when calling the Cardano API wrapper.

Reply:
> Perhaps it won't be necessary to tidy though if we can work out why the slowdown happens when calling the Cardano API wrapper.

👍 Yeah, let's leave it. Besides going back to pure cardano-api it's also _possible_ that we realise we want to switch to ledger types also say after doing the balanceTx library switch.